### PR TITLE
Scenario 4

### PR DESCRIPTION
**Pull request by Polina-Gubina**
Closes: #92

### CODE REVIEW

File: delatore/configuration/resources/sources.yaml [hunk 1]
```diff
@@ -113,6 +113,25 @@
...
+- name: influxdb_autoscalling
+  params:
+    host: 'influx1.eco.tsi-dev.otc-service.com'
+    port: 8086
+    username: 'csm'
+    database: 'csm'
+    metrics:
+      - name: LB_AUTOSCALING
+        metric_id: autoscaling
+        query: "SELECT mean(\"elapsed\") AS \"Response time\" FROM \"as_loadbalancer\" GROUP BY time(10m) fill(linear) ORDER BY time DESC LIMIT 1"
```
Comment (anton-sidelnikov) [inline delatore/configuration/resources/sources.yaml:125]: try to use single quotes

File: delatore/sources/__init__.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 from .awx_api import AWXApiSource
 from .http import AWXWebHookSource
 from .influx import InfluxSource, InfluxSourceLBTiming, InfluxSourceLBDOWN, InfluxSourceLBDOWNFailCount, InfluxSourceDiskStateRead, \
-                                            InfluxSourceDiskStateWrite, InfluxSourceDiskStateReadSFS, InfluxSourceDiskStateWriteSFS
+                                            InfluxSourceDiskStateWrite, InfluxSourceDiskStateReadSFS, InfluxSourceDiskStateWriteSFS, InfluxSourceAutoscalling
```
Comment (outcatcher) [inline delatore/sources/__init__.py:9]: Please group imports with parentheses instead of `\` multilining

File: delatore/sources/influx.py [hunk 3]
```diff
@@ -577,6 +577,64 @@ async def _get_status(self, metric):
...
+
+
+    async def _get_status(self, metric):
+        query = metric.query.format(entity=metric.metric_id)
+        last_record = await self.influx_client.query(query)
+        last_time_ms = None
+        error_status = False
+        status = Status.OK
+        try:
+            response_time = last_record.raw['series'][0]['values'][0][1]
```
Comment (outcatcher) [inline delatore/sources/influx.py:624]: `last_record.raw['series'][0]['values'][0]` is common for two expressions, you can move it into varialbe (and handle exceptions more explicit)

File: tests/02_acceptance/21_sources_ext/test_influx.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 from apubsub.client import Client
 
 from delatore.sources import InfluxSource, InfluxSourceLBTiming, InfluxSourceLBDOWN, InfluxSourceLBDOWNFailCount, \
-                             InfluxSourceDiskStateRead, InfluxSourceDiskStateWrite, InfluxSourceDiskStateReadSFS, InfluxSourceDiskStateWriteSFS
+                             InfluxSourceDiskStateRead, InfluxSourceDiskStateWrite, InfluxSourceDiskStateReadSFS, InfluxSourceDiskStateWriteSFS,  InfluxSourceDiskStateWriteSFS, InfluxSourceAutoscalling
```
Comment (outcatcher) [inline tests/02_acceptance/21_sources_ext/test_influx.py:7]: Multiline can be in parentheses

File: tests/02_acceptance/conftest.py [hunk 5]
```diff
@@ -3,7 +3,7 @@
 import pytest
 
 from delatore.configuration import DEFAULT_INSTANCE_CONFIG
-from delatore.sources import AWXApiSource, InfluxSource, InfluxSourceLBTiming, InfluxSourceLBDOWN, InfluxSourceLBDOWNFailCount, InfluxSourceDiskStateRead, InfluxSourceDiskStateWrite,InfluxSourceDiskStateReadSFS, InfluxSourceDiskStateWriteSFS
+from delatore.sources import AWXApiSource, InfluxSource, InfluxSourceLBTiming, InfluxSourceLBDOWN, InfluxSourceLBDOWNFailCount, InfluxSourceDiskStateRead, InfluxSourceDiskStateWrite,InfluxSourceDiskStateReadSFS, InfluxSourceDiskStateWriteSFS, InfluxSourceAutoscalling
```
Comment (outcatcher) [inline tests/02_acceptance/conftest.py:6]: Seems like imports are not sorted

File: delatore/configuration/resources/sources.yaml [hunk 6]
```diff
@@ -113,6 +113,25 @@
...
+    username: 'csm'
+    database: 'csm'
+    metrics:
+      - name: LB_AUTOSCALING
+        metric_id: as_loadbalancer
+        query: 'SELECT mean(elapsed) FROM {entity} GROUP BY time(10m) fill(linear) ORDER BY time DESC LIMIT 1'
+        timeout: 60
+      - name: LB_AUTOSCALING:CPU utilization
+        metric_id: cpu
+        query: 'SELECT max(usage_system)  FROM cpu  WHERE host =~ /scn4/ ORDER BY time DESC LIMIT 1'
```
Comment (lego963) [inline delatore/configuration/resources/sources.yaml:129]: FROM {entity}

File: delatore/configuration/resources/sources.yaml [hunk 7]
```diff
@@ -113,6 +113,25 @@
   timings:
     polling_interval: 60
     request_timeout: 25
+- name: influxdb_autoscalling
```
Comment (lego963) [inline delatore/configuration/resources/sources.yaml:116]: misspelling `influxdb_autoscaling`

File: delatore/configuration/resources/sources.yaml [hunk 8]
```diff
@@ -113,6 +113,25 @@
...
+        query: 'SELECT mean(elapsed) FROM {entity} GROUP BY time(10m) fill(linear) ORDER BY time DESC LIMIT 1'
+        timeout: 60
+      - name: LB_AUTOSCALING:CPU utilization
+        metric_id: cpu
+        query: 'SELECT max(usage_system)  FROM cpu  WHERE host =~ /scn4/ ORDER BY time DESC LIMIT 1'
+        timeout: 60
+  topic_prefix: 'INFLUX_AUTOSCALLING'
+  timings:
+    polling_interval: 60
+    request_timeout: 25
```
Comment (lego963) [inline delatore/configuration/resources/sources.yaml:134]: too big `request_timeout`, set to 5

File: delatore/sources/influx.py [hunk 9]
```diff
@@ -577,6 +577,65 @@ async def _get_status(self, metric):
         return disk_statuses
 
 
+
+class InfluxSourceAutoscalling(InfluxSource):
+    """InfluxSourceAutoscallingclient"""
+
+    # pylint: disable=unused-variable
```
Comment (lego963) [inline delatore/sources/influx.py:584]: redendant

File: delatore/sources/influx.py [hunk 10]
```diff
@@ -577,6 +577,65 @@ async def _get_status(self, metric):
...
+        last_time_ms = None
+        error_status = False
+        status = Status.OK
+        try:
+            response_line = last_record.raw['series'][0]['values'][0]
+        except(IndexError, KeyError):
+            status = Status.NO_DATA
+            return error_status, last_time_ms, status
+        response_time = response_line[1]
+        if response_time > self._threshold:
```
Comment (lego963) [inline delatore/sources/influx.py:628]: in _get_status we get info
in _get_update we compare info from _get_status with thresholds or other params

**Pull request closed without merging**